src/dc_receive_imf.rs

```diff
@@ -864,9 +864,10 @@ async fn add_parts(
         DC_CHAT_ID_TRASH
     });
 
-    // Extract ephemeral timer from the message.
-    let mut ephemeral_timer = if let Some(value) = mime_parser.get_header(HeaderDef::EphemeralTimer)
-    {
+    // Extract ephemeral timer from the message or use the existing timer if the message is not fully downloaded.
+    let mut ephemeral_timer = if is_partial_download.is_some() {
+        chat_id.get_ephemeral_timer(context).await?
```

Comment:
This is assuming that the timer for the partial download is the same as the current timer, but the code updating the timer is not skipped. As a result, ephemeral timer timestamp will be updated, even though we did not receive any timer. If we then receive a late message actually updating the timer with a timestamp lower than the timestamp of the partially downloaded message, it will be erroneously ignored.

Reply:
>  but the code updating the timer is not skipped

it is skipped by the comparison `chat_id.get_ephemeral_timer(context).await? != ephemeral_timer` a few lines below.

but yes, maybe this is not good coding style :)